torch/_dynamo/variables/builder.py

```diff
@@ -1284,8 +1284,8 @@ def _clone_input(value):
 
         # Handle recursive calls here
         elif (
```

Comment:
this line was not working properly because in https://github.com/pytorch/pytorch/pull/105308, fakification of tensor subclasses results in the parent class (the tensor subclass) being a regular tensor, and the children tensors being Fake tensors.

for now I'm just following existing design, would be curious to understand why parent tensor subclasses are not fakified

Reply:
Can you elaborate more?

> fakification of tensor subclasses results in the parent class (the tensor subclass) being a regular tensor, and the children tensors being Fake tensors.

This is a little hard to unpack for me, if the parent class is a tensor subclass, what are the children here - are you saying that the current program works for tensor subclasses with only 1 layer of depth, but for inheritance of k layers, we only fakify the lowest (child-est?) layer?

Reply:
here is my understanding of the relevant parts of the design of fakefying tensor subclasses introduced in  https://github.com/pytorch/pytorch/pull/105308

```
# before fakefying: a tensor subclass wraps 1 to N regular tensors
# example:
FooTensor(_data: torch.Tensor)

# after fakefying: a tensor subclass (not fake, type FooTensor), wraps 1 to N fake tensors
# example:
FooTensor(_data: FakeTensor)
```

in this design, checking `isinstance(t, FakeTensor)` seems incorrect because that does not work correctly for tensor subclasses.  The proper way to do it is to use the `is_fake` function introduced in that PR.

wdyt?

Reply:
> This is a little hard to unpack for me, if the parent class is a tensor subclass, what are the children here - are you saying that the current program works for tensor subclasses with only 1 layer of depth, but for inheritance of k layers, we only fakify the lowest (child-est?) layer?

From reading https://github.com/pytorch/pytorch/pull/105308/files, arbitrary depth should be handled.  However, it's not something I have tested.